frontend/flashfolio/src/Viewer.js

```diff
@@ -79,19 +79,43 @@ export default function Viewer({viewMode="view"}) {
 	useEffect(() => {
 		if (isInitialMount.current) {
 			isInitialMount.current = false;
-		} 
+		}
 		else {
 			/* useEffect code here to be run on count update only */
-			if(cardIterator < flashdeck.current.Cards.length) {
-				if((!shufOn))
+			if (cardIterator < flashdeck.current.Cards.length) {
+				if ((!shufOn))
 					setFlashcard(flashdeck.current.Cards[cardIterator]);
 				else
-					setFlashcard(flashdeck.current.Cards[shufOrder[cardIterator]]);		
+					setFlashcard(flashdeck.current.Cards[shufOrder[cardIterator]]);
 			}
 			else { setCardIterator(0); }
 		}
 	}, [cardIterator])
 
+	function deleteCard() {
+		/* if there's 1 card, then make an empty card*/
+		if (flashdeck.current.Cards.length === 1) {
+			flashdeck.current.Cards[0] = {};
```

Comment:
The object will still need a `FrontSide` value and a `BackSide` value. You can just set these to the empty string.

Reply:
there is an issue when i try to make .FrontSide and .BackSide empty strings like this: 

```
if (flashdeck.current.Cards.length === 1) {
	flashdeck.current.Cards[0].FrontSide = "";
	flashdeck.current.Cards[0].BackSide = "";
	/* view the blank card */
	setFlashcard(flashdeck.current.Cards[cardIterator]);
	return;
	}
```

with this style, there is a bug that occurs when you delete the last card: the view will not update to the blank card. instead, the view will remain on the last card. however, when the entire card is set to {}, how I've written it originally, the view appears properly blank upon deleting without any errors.

i will look further into how to make this work with empty FrontSide and BackSide attributes

Reply:
try

```js
flashdeck.current.Cards[0] = {};
flashdeck.current.Cards[0].FrontSide = "";
flashdeck.current.Cards[0].BackSide = "";
```

Reply:
I have made the requested change